### PR TITLE
Reply message full feature

### LINKED ISSUE #47: Reply messages
Add the feature: reply messages.
According to #22

### PR DESCRIPTION
**Pull request by GustavoMeza**
Added the reply message feature.
Full functionality with tests.
This closes #47

### CODE REVIEW

File: src/test/java/codeu/model/store/basic/MessageStoreTest.java [hunk 1]
```diff
@@ -92,12 +96,4 @@ public void testGetMessage_ById_notFound() {
 
     Assert.assertNull(message);
   }
-
```
Comment (khankuan) [inline src/test/java/codeu/model/store/basic/MessageStoreTest.java]: Good! Always good to do minor refactoring as you code new features :)

File: src/main/webapp/WEB-INF/view/chat.jsp [hunk 2]
```diff
@@ -38,16 +61,55 @@ List<Message> messages = (List<Message>) request.getAttribute("messages");
 
   <script>
     // scroll the chat div to the bottom
+    var idReply = null;
```
[the remark this answers is not in the document]
Comment (GustavoMeza) [reply src/main/webapp/WEB-INF/view/chat.jsp]: Yes, you're right 😞

File: src/main/webapp/WEB-INF/view/chat.jsp [hunk 3]
```diff
@@ -38,16 +61,55 @@ List<Message> messages = (List<Message>) request.getAttribute("messages");
...
+
     function scrollChat() {
       var chatDiv = document.getElementById('chat');
       chatDiv.scrollTop = chatDiv.scrollHeight;
-    };
+    }
+
+    function reply(id, html) {
```
[the remark this answers is not in the document]
Comment (GustavoMeza) [reply src/main/webapp/WEB-INF/view/chat.jsp]: I agree! I'll change this.

File: src/main/webapp/WEB-INF/view/chat.jsp [hunk 4]
```diff
@@ -59,21 +121,33 @@ List<Message> messages = (List<Message>) request.getAttribute("messages");
...
+        if(message.getParentId() != null) {
+          out.print("<div class=\"parent\">");
+          Message parent = MessageStore.getInstance().getMessage(message.getParentId());
+          out.print(formatChatMessage(parent));
+          out.print("</div>");
+          out.print("&#8618;");
+        }
+        out.print(formatChatMessage(message));
+        String replyButton =String.format(
+            "<button onclick='reply(\"%s\", \"%s\")' class='transparent'>&#8618;</button>",
```
[the remark this answers is not in the document]
Comment (GustavoMeza) [reply src/main/webapp/WEB-INF/view/chat.jsp]: So you think the button should be created on hover?

File: src/main/webapp/WEB-INF/view/chat.jsp [hunk 5]
```diff
@@ -38,16 +61,55 @@ List<Message> messages = (List<Message>) request.getAttribute("messages");
...
   <%@ include file="../component/navbar.jsp" %>
 
+  <%!
+    public String formatChatMessage(Message message) {
+      return String.format("%s: %s",
+              formatUserName(message.getAuthorId()),
+              message.getContent());
+    }
+
+    public String formatHtml(String string) {
```
[the remark this answers is not in the document]
Comment (GustavoMeza) [reply src/main/webapp/WEB-INF/view/chat.jsp]: Ok!

File: src/main/webapp/WEB-INF/view/chat.jsp (hunk 3 above)
[the remark this answers is not in the document]
Comment (khankuan) [reply src/main/webapp/WEB-INF/view/chat.jsp]: That is a good point. Let's leave it that way then. `formatChatReply` definitely would be neater. Thanks.

[the remark this answers is not in the document]
Comment (GustavoMeza) [reply src/main/webapp/WEB-INF/view/chat.jsp]: I did some changes, only some code quality improvements.
Added comments and reorganized view code.

**Pull request merged**